lib/FileUploadDataSource.ts

```diff
@@ -112,22 +174,10 @@ export default class FileUploadDataSource extends RemoteGraphQLDataSource {
 
     // This must come before the file contents append bellow
     form.append('map', JSON.stringify(fileMap));
-
-    resolvedFiles.forEach(
-      ({ createReadStream, filename, mimetype: contentType }, i: number) => {
-        form.append(i.toString(), createReadStream(), {
-          contentType,
-          filename,
-          /*
-              Set knownLength to NaN so node-fetch does not set the
-              Content-Length header and properly set the enconding
-              to chunked.
-              https://github.com/form-data/form-data/pull/397#issuecomment-471976669
-          */
-          knownLength: Number.NaN,
-        });
-      },
-    );
+    const handler = this.useChunkedTransfer
+      ? FileUploadDataSource.addChunkedDataToForm
+      : FileUploadDataSource.addDataToForm;
+    await Promise.all(resolvedFiles.map(handler(form)));
```

Comment:
and actually we could move one level up, since in the chunked case we're now creating useless promises and waiting them to resolve. Make the resulting handler take an array instead of a single item. Then in one you do the `forEach()`, in the other you do the `Promise.all(resolvedFiles.map(...))`